### PR TITLE
SPLAT-1129: create fixtures for VSphereProviderSpec zones

### PR DESCRIPTION
**Pull request by rvanderp3**

### CODE REVIEW

File: testutils/resourcebuilder/config/v1/infrastructure.go [hunk 1]
```diff
@@ -161,6 +161,96 @@ func (i InfrastructureBuilder) AsOpenStack(name string) InfrastructureBuilder {
...
+
+// AsVSphereWithFailureDomains returns a VSphere infrastructure resource with failure domains.
+// if failureDomains = nil, default failure domains will be applied to the resource which are
+// compatible with machinev1beta1resourcebuilder default failure domain names.
+func (i InfrastructureBuilder) AsVSphereWithFailureDomains(name string, failureDomains *[]configv1.VSpherePlatformFailureDomainSpec) InfrastructureBuilder {
+	infraBuilder := i.AsVSphere(name)
+	if failureDomains != nil {
+		infraBuilder.spec.PlatformSpec.VSphere.FailureDomains = *failureDomains
+	} else {
+		infraBuilder.spec.PlatformSpec.VSphere.FailureDomains = []configv1.VSpherePlatformFailureDomainSpec{
```
Comment (elmiko) [inline testutils/resourcebuilder/config/v1/infrastructure.go:199]: does it make sense to use the failure domain builder here?

**Review by elmiko — COMMENTED** [review]
this makes sense to me, just a quick question

**Review by JoelSpeed — COMMENTED** [review]
/lgtm
/approve

**Comment by rvanderp3** [comment]
/hold cancel

discussed with Mike and this is ready to go.

**Pull request merged**